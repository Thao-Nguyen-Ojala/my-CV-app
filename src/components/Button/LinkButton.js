import React from 'react'
import Button from './Button';
import { Link } from 'react-router-dom';


function LinkButton({children, type, onClick, buttonStyle, buttonSize, buttonColor, directTo}) {

  return (
    <div>
      <Link to={`/${directTo}`}>
        <Button
          buttonColor={buttonColor}
          buttonSize={buttonSize}
          buttonStyle={buttonStyle}
          onClick={onClick}
          type={type}
          children={children}
        />
      </Link>
    </div>
  )
}

export default LinkButton
