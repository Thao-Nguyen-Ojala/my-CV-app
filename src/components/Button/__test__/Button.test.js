import React from 'react';
import ReactDOM from 'react-dom'
import Button from '../Button'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('render Button without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button>Button Testing</Button>, div)
})

it('render button correctly', () => {
  const {getByTestId} = render(<Button>INTRODUCTION</Button>)
  expect(getByTestId('button')).toHaveTextContent('INTRODUCTION')
})

it('matches Button snapshot', () => {
  const testingButton = renderer.create(<Button className='btns' buttonStyle='btn--outline' buttonSize='btn--large'  buttonColor='blue'u>Testing</Button>).toJSON();
  expect(testingButton).toMatchSnapshot()
})



