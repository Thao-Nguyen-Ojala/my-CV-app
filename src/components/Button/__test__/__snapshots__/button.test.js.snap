// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Button snapshot 1`] = `
<button
  className="btn btn--outline btn--large blue"
  data-testid="button"
>
  Testing
</button>
`;
